catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandlerTest.java

```diff
@@ -0,0 +1,207 @@
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyMap;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import ddf.catalog.data.BinaryContent;
+import ddf.catalog.data.impl.BinaryContentImpl;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.transform.QueryResponseTransformer;
+import java.io.ByteArrayInputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.activation.MimeType;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.HttpHeaders;
+import org.boon.Boon;
+import org.codice.ddf.catalog.ui.query.cql.CqlQueryResponse;
+import org.codice.ddf.catalog.ui.query.cql.CqlRequest;
+import org.codice.ddf.catalog.ui.util.EndpointUtil;
+import org.eclipse.jetty.http.HttpStatus;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+import spark.Request;
+import spark.Response;
+
+public class CqlTransformHandlerTest {
+
+  private List<ServiceReference> queryResponseTransformers;
+  private CqlTransformHandler cqlTransformHandler;
+  private BinaryContent binaryContent;
+
+  @Mock private ServiceReference<QueryResponseTransformer> mockQueryResponseTransformer;
+  @Mock private BundleContext mockBundleContext;
+  @Mock private EndpointUtil mockEndpointUtil;
+  @Mock private Request mockRequest;
+  @Mock private CqlQueryResponse mockCqlQueryResponse;
+  @Mock private QueryResponse mockQueryResponse;
+  @Mock private QueryResponseTransformer mockServiceReference;
+  @Mock private ServletOutputStream mockServletOutputStream;
+  @Mock private HttpServletResponse mockHttpServiceResponse;
+
+  private static final String GZIP = "gzip";
+  private static final String NO_GZIP = "";
+  private static final String QUERY_PARAM = ":transformerId";
+  private static final String RETURN_ID = "kml";
+  private static final String OTHER_RETURN_ID = "xml";
+  private static final String MIME_TYPE = "application/vnd.google-earth.kml+xml";
+  private static final String SAFE_BODY =
+      "{\"src\":\"ddf.distribution\",\"start\":1,\"count\":250,\"cql\":\"anyText ILIKE '*'\",\"sorts\":[{\"attribute\":\"modified\",\"direction\":\"descending\"}],\"id\":\"7a491439-948e-431b-815e-a04f32fecec9\"}";
+  private static final String CONTENT = "test";
+  private static final String SERVICE_NOT_FOUND = "{\"message\":\"Service not found\"}";
+  private static final String SERVICE_SUCCESS = Boon.toJson("");
+  private static final String ATTACHMENT_REGEX =
+      "^attachment;filename=export-\\d\\d\\d\\d-\\d\\d-\\d\\dT\\d\\d:\\d\\d:\\d\\d\\.\\d\\d\\dZ."
```

Comment:
✏️ Couple things here
1) If you want you can specify a greedy quantifier To match something X number of times, as `{X}`. So instead of `\\d\\d\\d\\d` you can do `\\d{4}`. Check out `Greedy Quantifiers` in the [javadoc for Pattern](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html) So this could be shortened a little.
2) if the regex there was an arbitrary iso8601 time then it gets [really, really advanced to actually cover most the cases](https://stackoverflow.com/a/28022901)... but since this is just coming from an `Instant.now().toString` it will always actually be an iso8601 formatted as above. 
3) if this wasn't a test class ( I mean you still could if you want to ) I'd recommend compiling the Pattern so that it isn't recompiled every call to String::matches.